src/javascript/SelectorTypes/registerSelectorTypesOnChange.js

```diff
@@ -2,9 +2,27 @@ import {getFields} from '~/utils/fields.utils';
 import {FieldConstraints} from './registerSelectorTypesOnChange.gql-queries';
 
 function arrayEquals(arr1, arr2) {
-    return arr1.length === arr2.length && arr1.every((value, index) => value === arr2[index]);
+    return arr1.length === arr2.length && arr1.every((value, index) => {
+        if (isObject(value)) {
+            return JSON.stringify(value) === JSON.stringify(arr2[index]);
```

Comment:
this can return false if order of attr is not the same, or at least not guaranteed e.g. `JSON.stringify({x:1, y:2}) ` is not equal to `JSON.stringify({y:2, x:1})`. I'm not sure if it matters here or not.

Reply:
I know but it will not matter here as everything will be in same order